searchcore/src/vespa/searchcore/proton/matching/query.cpp

```diff
@@ -64,11 +64,14 @@ find_location_terms(Node *tree) {
     std::vector<ProtonLocationTerm *> retval;
     std::vector<Node *> nodes;
     nodes.push_back(tree);
+    // Note the nodes vector being iterated over is appended in the loop
     for (size_t i = 0; i < nodes.size(); ++i) {
-        if (auto loc = dynamic_cast<ProtonLocationTerm *>(nodes[i])) {
+        Node * node = nodes[i];
+        if (auto loc = dynamic_cast<ProtonLocationTerm *>(node)) {
```

Comment:
if you are looking for only a single specific class (ensure no sub-classes by making it final), using std::type_index(typeid(...)) is much faster than dynamic cast (and even faster than virtual predicates combined with static cast)

Reply:
hmm, that one is new to me. Will take a look.

Reply:
various strategies are tested in 'vespa/vespalib/src/tests/detect_type_benchmark'